### ISSUE TITLE
2D shapes demo renders all shapes with same color on Dx12 backend

### ISSUE THREAD

**Issue by ByteNybbler** [bevyengine/bevy#12741]
## Bevy version

v0.13.1

## Relevant system information

- cargo 1.77.0 (3fe68eabf 2024-02-29)
- Windows 10
- `AdapterInfo { name: "Microsoft Basic Render Driver", vendor: 5140, device: 140, device_type: Cpu, driver: "", driver_info: "", backend: Dx12 }`
- `bevy_diagnostic::system_information_diagnostics_plugin::internal: SystemInfo { os: "Windows 10 Home", kernel: 
"19045", cpu: "Intel(R) Core(TM) i7-4790 CPU @ 3.60GHz", core_count: "4", memory: "15.9 GiB" }`

## What you did

`cargo run --example 2d_shapes`

## What went wrong

All shapes are rendered with the same color, and the color seems to be non-deterministically chosen each time the example runs:

![image](https://github.com/bevyengine/bevy/assets/22783581/2d1972ff-eb27-48c2-97ee-c5ae63e62dd1)

## Additional information

For whatever reason, Bevy 0.13.1 appears to be falling back to using CPU rendering instead of using my integrated GPU. That wasn't the case with Bevy 0.12.

**Comment by SolarLiner** [comment]
Are your graphics drivers up to date? This kind of "reverting back to CPU rendering" is often due to a faulty GPU driver that hasn't completed its update.

**Comment by ByteNybbler** [comment]
> Are your graphics drivers up to date? This kind of "reverting back to CPU rendering" is often due to a faulty GPU driver that hasn't completed its update. What's your GPU?

My GPU driver is Intel(R) HD Graphics 4600 version 20.19.15.4531 dated 9/29/2016. It's old hardware.

I'm not sure if the driver is up to date so I've been looking around, but I've noticed something else in the meantime.

Running the 2D shapes example in bevy v0.12.0 properly uses my Intel GPU: `AdapterInfo { name: "Intel(R) HD Graphics 4600", vendor: 32902, device: 1042, device_type: IntegratedGpu, driver: "", driver_info: "", backend: Dx12 }`

But running the 2D shapes example in bevy v0.13.1 uses CPU rendering instead: `AdapterInfo { name: "Microsoft Basic Render Driver", vendor: 5140, device: 140, device_type: Cpu, driver: "", driver_info: "", backend: Dx12 }`

I'm still in the process of learning about wgpu, but I wonder if it's possible that a recent version of bevy has started requiring more GPU features and hence stopped allowing my GPU.

**Comment by mockersf** [comment]
this is running in CI on the software renderer and every shape has the expected color
```
bevy_diagnostic::system_information_diagnostics_plugin::internal: SystemInfo { os: "Windows Server 2022 Datacenter", kernel: "20348", cpu: "AMD EPYC 7763 64-Core Processor", core_count: "2", memory: "16.0 GiB" }
bevy_render::renderer: AdapterInfo { name: "Microsoft Basic Render Driver", vendor: 5140, device: 140, device_type: Cpu, driver: "", driver_info: "", backend: Dx12 }
```

**Comment by janhohenheim** [comment]
Pretty sure this was resolved by the wgpu 0.20 update. Feel free to reopen if that's not the case :)
